ansible_collections/arista/avd/plugins/plugin_utils/topology_generator_utils.py

```diff
@@ -0,0 +1,715 @@
+    Args:
+        graph: dictionary with node and its neighbors list
+        start_node: Starting point/root node of the graph
+        node_list: List of nodes
+    Returns:
+         dict: Return a dictionary with level as key and node list as value
+    """
+
+    # array to store level of each node
+    node_level_dict = {}
+    marked = {node: False for node in node_list}
+
+    # create a queue
+    que = queue.Queue()
+    # enqueue element x
+    que.put(start_node)
+    # initialize level of start_node
+    # node to 0
+    node_level_dict[start_node] = 0
+    # marked it as visited
+    marked[start_node] = True
+    # do until queue is empty
+    while not que.empty():
+        # get the first element of queue
+        start_node = que.get()      
+        # traverse neighbors of node start_node
+        if start_node in graph:  #key checking
+            for i in graph[start_node]:
+                # neighbor is neighbor of node start_node
+                neighbor = i["neighborDevice"] 
+                  
+                # if neighbor is not marked already
+                if neighbor in marked.keys() and not marked[neighbor]:
+                    # enqueue neighbor in queue       
+                    que.put(neighbor)
+
+                    # level of neighbor is level of start_node + 1
+                    node_level_dict[neighbor] = node_level_dict[start_node] + 1
+
+                    # mark neighbor
+                    marked[neighbor] = True
+ 
+    level_dict = {}
+    for node, level in node_level_dict.items():
+        if level not in level_dict:
+            level_dict[level] = [node]
+        else:
+            level_dict[level].append(node)
+
+    return level_dict, node_level_dict
+
+def generate_topology_hampton(destination, old_level_dict, node_neighbor_dict, ol, undefined_rank_nodes, node_port_val):
+    """
+    Generate topology diagram using node neighbour dictionary
+
+    Args:
+      destination: file name in which topology diagram get saved
+      old_level_dict: List of nodes and its corrensponding level
+                        ex- {0: ['0'], 1: ['dc1-ss1', 'dc1-ss2']}
+      node_neighbor_dict: Dictionary with nodes and neighbors list and their ports
+                        ex- {'0': [{'nodePort': '', 'neighborDevice': 'dc2-ss3', 'neighborPort': ''}]
+      ol: output_list contain dictionary having cluster name and having their corresponding nodes 
+      undefined_rank_nodes: nodes which dont have parent
+      node_port_val: port values for each node
+    """ 
+    # Remove psuedo 0'th element
+    del old_level_dict[0]
+
+    # Make a level loopup dict
+    level_dict = {}
+    for idx, nodes in old_level_dict.items():
+        for node in nodes:
+            level_dict[node] = idx
+
+
+    ol = subgroup_inventor_recursive(level_dict, ol[0])
+
+    ol = calculate_box_size_recursive(level_dict, ol)
+
+
+    GROUPOFFSET = LVLSPACING
```

Comment:
why is `GROUPOFFSET` capitalized? Please keep capitalized variable names for globals